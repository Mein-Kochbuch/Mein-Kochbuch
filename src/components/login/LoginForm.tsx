import React, {useState} from "react";
import {Linking} from "react-native";
import InputField from "./InputField";
import styled from "styled-components/native/dist/styled-components.native.esm";
import {Colors} from "react-native/Libraries/NewAppScreen";
import {useColorScheme} from "react-native-appearance";

export default function LoginForm({onSubmit}) {
    const [credentials, setCredentials] = useState({})

    const isDarkMode = useColorScheme() === 'dark';

    const onEmailChange = (value) => {
        setCredentials((oldCredentials) => {
            return {...oldCredentials, "username": value}
        })
    }

    const onPasswordChange = (value) => {
        setCredentials((oldCredentials) => {
            return {...oldCredentials, "password": value}
        })
    }

    return (
        <StyledView>
            <InputField
                title={"E-Mail"}
                onChangeText={onEmailChange}
                placeholder="E-Mail"
                value={credentials.username}
                keyboardType={"email-address"}/>
            <InputField
                title={"Password"}
                onChangeText={onPasswordChange}
                placeholder="Password"
                value={credentials.password}
                secureTextEntry={true}/>

            <PrivacyBanner>
                Wenn du fortfährst, stimmst du den
                <LinkText onPress={() => Linking.openURL('https://mein-kochbuch.org/agb/')}> AGBs </LinkText>
                zu. In unseren
                <LinkText onPress={() => Linking.openURL('https://mein-kochbuch.org/privacypolicy/')}> Privacy
                    Policy </LinkText>
                erfährst du, wie wir deine Daten erfassen, verwenden und teilen.
            </PrivacyBanner>

            <StyledTouchableOpacity isDarkMode={isDarkMode} onPress={() => onSubmit(credentials)}>
                <StyledText>
                    Login
                </StyledText>
            </StyledTouchableOpacity>
        </StyledView>
    )
}

const StyledView = styled.View`
  display: flex;
  align-items: center;
  flex-direction: column;
`

const LinkText = styled.Text`
  color: #0074D9;
  text-decoration-line: underline;
`

const PrivacyBanner = styled.Text`
  margin: 24px;
  text-align: center;
`

const StyledText = styled.Text`
  text-align: center;
  font-size: 30px;
`

const StyledTouchableOpacity = styled.TouchableOpacity`
  margin: 12px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  width: 80%;
  min-width: 300px;
  border-style: solid;
  border-color: ${props => props.isDarkMode ? Colors.lighter : Colors.darker};
  border-radius: 10px;
  border-width: 2px;
`

