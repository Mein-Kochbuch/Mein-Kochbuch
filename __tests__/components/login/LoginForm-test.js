import React from "react";
import renderer from "react-test-renderer";
import LoginForm from "../../../src/components/login/LoginForm";
import {useColorScheme} from "react-native-appearance";
import {fireEvent, render} from "@testing-library/react-native";

jest.mock("../../../src/components/recipes/RecipeListItem", () => {
    return (props) => {
        return <div {...props}/>
    }
});

jest.mock("react-native-appearance", () => ({
    useColorScheme: jest.fn(),
}));


describe('LoginForm Test', () => {

    afterEach(() => {
        jest.clearAllMocks();
    });


    it('LoginForm Snapshot', () => {
        useColorScheme.mockReturnValueOnce('dark');
        const onSubmit = jest.fn();

        const component = renderer.create(
            <LoginForm onSubmit={onSubmit}/>
        );

        expect(component.toJSON()).toMatchSnapshot()
    })

    it('LoginForm Login Button press', () => {
        useColorScheme.mockReturnValueOnce('dark');
        const onSubmit = jest.fn();

        const {getByText} = render(
            <LoginForm onSubmit={onSubmit}/>
        );

        fireEvent(getByText('E-Mail'), 'onChangeText', "testMail");
        fireEvent(getByText('Password'), 'onChangeText', "testPassword");

        fireEvent(getByText('Login'), 'onPress');
        expect(onSubmit).toBeCalledWith({username: "testMail", password: "testPassword"})
    })
})
