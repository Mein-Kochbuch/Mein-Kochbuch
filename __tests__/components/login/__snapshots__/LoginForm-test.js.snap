// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm Test LoginForm Snapshot dark 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "marginBottom": 12,
          "marginLeft": 12,
          "marginRight": 12,
          "marginTop": 12,
        },
      ]
    }
  >
    <Text>
      E-Mail
    </Text>
    <TextInput
      isDarkMode={false}
      keyboardType="email-address"
      onChangeText={[Function]}
      placeholder="E-Mail"
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderColor": "#222",
            "borderRadius": 10,
            "borderStyle": "solid",
            "borderWidth": 2,
            "display": "flex",
            "flexDirection": "column",
            "justifyContent": "space-between",
            "minWidth": 300,
            "width": "80%",
          },
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "marginBottom": 12,
          "marginLeft": 12,
          "marginRight": 12,
          "marginTop": 12,
        },
      ]
    }
  >
    <Text>
      Password
    </Text>
    <TextInput
      isDarkMode={false}
      onChangeText={[Function]}
      placeholder="Password"
      secureTextEntry={true}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderColor": "#222",
            "borderRadius": 10,
            "borderStyle": "solid",
            "borderWidth": 2,
            "display": "flex",
            "flexDirection": "column",
            "justifyContent": "space-between",
            "minWidth": 300,
            "width": "80%",
          },
        ]
      }
    />
  </View>
  <Text
    style={
      Array [
        Object {
          "marginBottom": 24,
          "marginLeft": 24,
          "marginRight": 24,
          "marginTop": 24,
          "textAlign": "center",
        },
      ]
    }
  >
    Wenn du fortf채hrst, stimmst du den
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#0074D9",
            "textDecorationLine": "underline",
          },
        ]
      }
    >
       AGBs 
    </Text>
    zu. In unseren
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#0074D9",
            "textDecorationLine": "underline",
          },
        ]
      }
    >
       Privacy Policy 
    </Text>
    erf채hrst du, wie wir deine Daten erfassen, verwenden und teilen.
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#F3F3F3",
        "borderRadius": 10,
        "borderStyle": "solid",
        "borderWidth": 2,
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "space-between",
        "marginBottom": 12,
        "marginLeft": 12,
        "marginRight": 12,
        "marginTop": 12,
        "minWidth": 300,
        "opacity": 1,
        "width": "80%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 30,
            "textAlign": "center",
          },
        ]
      }
    >
      Login
    </Text>
  </View>
</View>
`;

exports[`LoginForm Test LoginForm Snapshot light 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "marginBottom": 12,
          "marginLeft": 12,
          "marginRight": 12,
          "marginTop": 12,
        },
      ]
    }
  >
    <Text>
      E-Mail
    </Text>
    <TextInput
      isDarkMode={false}
      keyboardType="email-address"
      onChangeText={[Function]}
      placeholder="E-Mail"
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderColor": "#222",
            "borderRadius": 10,
            "borderStyle": "solid",
            "borderWidth": 2,
            "display": "flex",
            "flexDirection": "column",
            "justifyContent": "space-between",
            "minWidth": 300,
            "width": "80%",
          },
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "marginBottom": 12,
          "marginLeft": 12,
          "marginRight": 12,
          "marginTop": 12,
        },
      ]
    }
  >
    <Text>
      Password
    </Text>
    <TextInput
      isDarkMode={false}
      onChangeText={[Function]}
      placeholder="Password"
      secureTextEntry={true}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderColor": "#222",
            "borderRadius": 10,
            "borderStyle": "solid",
            "borderWidth": 2,
            "display": "flex",
            "flexDirection": "column",
            "justifyContent": "space-between",
            "minWidth": 300,
            "width": "80%",
          },
        ]
      }
    />
  </View>
  <Text
    style={
      Array [
        Object {
          "marginBottom": 24,
          "marginLeft": 24,
          "marginRight": 24,
          "marginTop": 24,
          "textAlign": "center",
        },
      ]
    }
  >
    Wenn du fortf채hrst, stimmst du den
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#0074D9",
            "textDecorationLine": "underline",
          },
        ]
      }
    >
       AGBs 
    </Text>
    zu. In unseren
    <Text
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "#0074D9",
            "textDecorationLine": "underline",
          },
        ]
      }
    >
       Privacy Policy 
    </Text>
    erf채hrst du, wie wir deine Daten erfassen, verwenden und teilen.
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#222",
        "borderRadius": 10,
        "borderStyle": "solid",
        "borderWidth": 2,
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "space-between",
        "marginBottom": 12,
        "marginLeft": 12,
        "marginRight": 12,
        "marginTop": 12,
        "minWidth": 300,
        "opacity": 1,
        "width": "80%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 30,
            "textAlign": "center",
          },
        ]
      }
    >
      Login
    </Text>
  </View>
</View>
`;
